### DISCUSSION TITLE
Add tutorials to hono.dev

### DISCUSSION THREAD

**Discussion by ghost** [honojs/hono#2969] [Ideas]
It would make a lot of sense to provide an official tutorial that presents the various APIs and middleware and recommended practices (Testing, file structure, etc.) for those interested in Hono.

My ideas:
- **Small**: To-do app
  - Routing, dynamic routing, validation, JSX
- **Medium**: chat app
  - Auth (JWT), websockets, RPC, Client components
- **Medium**: Petstore app
  - Testing, OpenAPI, Database (Cloudflare D1), Deployment (Cloudflare Workers)
  - From the famous [Swagger Petstore](https://petstore3.swagger.io/)

**Comment by ghost** [comment]
@yusukebe What do you think?

**Reply by yusukebe** [reply]
That sounds good. But increasing content means we will have more to maintain. So, we must be careful. Let's consider it after updating the structure of our website (we have some backlog to do it).

**Reply by tomByrer** [reply]
Many of us use examples as 'tutorials'.
If someone could collect any examples, videos, walk-though repos, that could fulfill much of the need without extra work on a Hono's core maintainers.

**Reply by adhanji8** [reply]
After the website structure updates, I'd be happy to contribute to this 👍

**Comment by ghost** [comment]
Hi @yusukebe @tomByrer @adhanji8 

Overall, I think a better way would be to provide community tutorial content (mainly videos) and official tutorials. And the official tutorial would have minimal maintenance costs if we add a test to make sure it works. @yusukebe Is this the right way to do it? I'm ready to start writing.
